src/test/java/org/opensearch/knn/index/FaissIT.java

```diff
@@ -267,6 +272,117 @@ public void testEndToEnd_whenMethodIsHNSWPQ_thenSucceed() {
         fail("Graphs are not getting evicted");
     }
 
+    @SneakyThrows
+    public void testHNSWSQFP16_whenIndexedAndQueried_thenSucceed() {
+        String indexName = "test-index-hnsw-sqfp16";
+        String fieldName = "test-field-hnsw-sqfp16";
+
+        KNNMethod hnswMethod = KNNEngine.FAISS.getMethod(KNNConstants.METHOD_HNSW);
+        SpaceType spaceType = SpaceType.L2;
```

Comment:
Could we also try IP just for sake of thoroughness?

Reply:
We can do 
```suggestion
        SpaceType[] spaceTypes = {SpaceType.L2, SpaceType.IP};
        Random random = new Random();
        SpaceType spaceType = spaceTypes[random.nextInt(spaceTypes.length)];
```

Reply:
updated